### PR TITLE
Documentation addition

### PR DESCRIPTION
**Pull request by aramando**
Just a small addition to the README to clarify that callbacks are accepted and Promises returned. I had to view source to determine this so would be good to merge in.

I also fixed the spelling of a few parameter names for good measure.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -48,6 +48,8 @@ these wrapper interfaces don't have the new parameters at the time we did the ch
 
 # JavaScript Usage
 
+**All the following methods accept optional success and error callbacks after all other available parameters, and also return a Promise object.**
```
Comment (victorsosa) [inline README.md]: can you remove that please?

Comment (aramando) [inline README.md]: I see what you mean; I've removed that part.

**Comment by aramando** [comment]
You're right; I've removed that part.

**Comment by victorsosa** [comment]
Thank you

**Pull request merged**